src/OptimalTransport.jl

```diff
@@ -784,4 +786,103 @@ function quadreg(mu, nu, C, ϵ; θ=0.1, tol=1e-5, maxiter=50, κ=0.5, δ=1e-5)
+respectively.
+
+A pre-computed optimal transport `plan` may be provided.
+
+See also: [`ot_plan`](@ref), [`emd2`](@ref)
+"""
+function ot_cost(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution; plan=nothing
+)
+    cost, _ = if plan === nothing
+        quadgk(0, 1) do q
+            return c(quantile(μ, q), quantile(ν, q))
+        end
+    else
+        quadgk(0, 1) do q
+            x = quantile(μ, q)
+            return c(x, plan(x))
+        end
+    end
+    return cost
+end
+
+"""
+    ot_plan(c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution)
+
+Compute the optimal transport plan for the Monge-Kantorovich problem with univariate
+distributions `μ` and `ν` as source and target marginals and cost function `c` of
+the form ``c(x, y) = h(|x - y|)`` where ``h`` is a convex function.
+
+In this setting, the optimal transport plan is the Monge map
+```math
+T = F_\\nu^{-1} \\circ F_\\mu
+```
+where ``F_\\mu`` is the cumulative distribution function of `μ` and ``F_\\nu^{-1}`` is the
+quantile function of `ν`.
+
+See also: [`ot_cost`](@ref), [`emd`](@ref)
+"""
+function ot_plan(c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution)
+    # Use T instead of γ to indicate that this is a Monge map.
+    T(x) = quantile(ν, cdf(μ, x))
+    return T
+end
+
+"""
+    ot_cost(c::SqEuclidean, μ::Union{Normal,MvNormal}, ν::Union{Normal,MvNormal})
+    
+Compute the squared 2-Wasserstein distance between multivariate Gaussian
+distributions `μ` and `ν` as source and target marginal.
+In this setting, the optimal transport cost can be computed as
+```math
+W_2^2(\\mu, \\nu) = \\|m_\\mu - m_\\nu \\|^2 + \\mathcal{B}(\\Sigma_\\mu, \\Sigma_\\nu)^2,
+```
+where ``\\mu = \\mathcal{N}(m_\\mu, \\Sigma_\\mu)``,
+``\\nu = \\mathcal{N}(m_\\nu, \\Sigma_\\nu)``, and ``\\mathcal{B}`` is the Bures metric.
+where ``\\mathcal{B}`` is the Bures metric.
+
+
+See also: [`ot_plan`](@ref), [`emd2`](@ref)
+"""
+function ot_cost(::SqEuclidean, μ::Union{Normal,MvNormal}, ν::Union{Normal,MvNormal})
+    return sqeuclidean(μ.μ, ν.μ) + sqbures(μ.Σ, ν.Σ)
+end
+
+"""
+    ot_plan(c::SqEuclidean, μ::Union{Normal,MvNormal}, ν::Union{Normal,MvNormal})
+
+Compute the optimal transport plan for the Monge-Kantorovich problem with multivariate
+normal distributions `μ` and `ν` as source and target marginals and cost function
+``c(x, y) = \\|x - y\\|_2^2``.
+In this setting, for ``\\mu = \\mathcal{N}(m_\\mu, \\Sigma_\\mu)`` and
+``\\nu = \\mathcal{N}(m_\\nu, \\Sigma_\\nu)``, the optimal transport plan is the Monge
+map
+```math
+T \\colon x \\mapsto m_\\nu
++ \\Sigma_\\mu^{-1/2}
+{\\big(\\Sigma_\\mu^{1/2} \\Sigma_\\nu \\Sigma_\\mu^{1/2}\\big)}^{1/2}\\Sigma_\\mu^{-1/2}
+(x - m_\\mu).
+"""
+function ot_plan(::SqEuclidean, μ::Union{Normal,MvNormal}, ν::Union{Normal,MvNormal})
+    Σμsqrt = μ.Σ^(-1 / 2)
+    T(x) = ν.μ + (Σμsqrt * sqrt(_gaussian_ot_A(μ.Σ, ν.Σ)) * Σμsqrt) * (x - μ.μ)
```

Comment:
Isn't the `sqrt` already included in `_gaussian_ot_A`?

Reply:
No, the `_gaussian_ot_A` returns the Σ μ^{1/2} *  Σ ν Σ μ ^{1/2}. So in the optimal map, I have to take the root of all of that.